notification/shoutrrr.go

```diff
@@ -2,14 +2,45 @@ package notification
 
 import (
 	"fmt"
+	"net/url"
+	"os"
 	"strings"
 
 	"github.com/containrrr/shoutrrr"
 	"github.com/containrrr/shoutrrr/pkg/types"
 	"github.com/flanksource/commons/collections"
+	"github.com/flanksource/commons/utils"
 	"github.com/flanksource/incident-commander/api"
+	"github.com/flanksource/incident-commander/mail"
 )
 
+// defaultSMTPPrefix indicates that the shoutrrr URL for smtp should use
+// the system's SMTP credentials.
+const defaultSMTPPrefix = "smtp://system/"
+
+// setSystemSMTPCredential modifies the shoutrrrURL to use the system's SMTP credentials.
+func setSystemSMTPCredential(shoutrrrURL string) (string, error) {
+	prefix := fmt.Sprintf("smtp://%s:%s@%s:%s/",
+		url.QueryEscape(os.Getenv("SMTP_USER")),
+		url.QueryEscape(os.Getenv("SMTP_PASSWORD")),
+		os.Getenv("SMTP_HOST"),
+		os.Getenv("SMTP_PORT"),
+	)
+	shoutrrrURL = strings.ReplaceAll(shoutrrrURL, defaultSMTPPrefix, prefix)
+
+	parsedURL, err := url.Parse(shoutrrrURL)
+	if err != nil {
+		return "", fmt.Errorf("failed to parse shoutrrr URL: %w", err)
+	}
+
+	query := parsedURL.Query()
+	query.Set("FromAddress", utils.Coalesce(mail.FromAddress, os.Getenv("SMTP_FROM"), os.Getenv("SMTP_USER")))
```

Comment:
Will this not always return mail.FromAddress as it has a default?

Reply:
on the rare cases where the flag is explicitly removed, this will handle it.